src/script.js

```diff
@@ -251,27 +250,55 @@ function isMultisigInput (script, allowIncomplete) {
   return chunks.slice(1).every(isCanonicalSignature)
 }
 
-function isMultisigOutput (script) {
-  var chunks = decompile(script)
-  if (chunks.length < 4) return false
-  if (chunks[chunks.length - 1] !== OPS.OP_CHECKMULTISIG) return false
-
-  var mOp = chunks[0]
-  var nOp = chunks[chunks.length - 2]
+function parseMultisigScript (scriptPubKey) {
+  var chunks = decompile(scriptPubKey)
+  if (chunks.length < 4) {
+    throw new Error('Multisig script should contain at least 4 elements')
+  }
+  if (chunks[chunks.length - 1] !== OPS.OP_CHECKMULTISIG) {
+    throw new Error('Final opcode should be OP_CHECKMULTISIG')
+  }
+  if (!types.Number(chunks[0])) {
+    throw new Error('First element must be a number-push opcode')
+  }
+  if (!types.Number(chunks[chunks.length - 2])) {
+    throw new Error('Second-last element must be a number-push opcode')
+  }
+  var m = chunks[0] - OP_INT_BASE
+  var n = chunks[chunks.length - 2] - OP_INT_BASE
 
-  if (!types.Number(mOp)) return false
-  if (!types.Number(nOp)) return false
+  if (m <= 0) {
+    throw new Error('Number of signatures required must be greater than zero')
+  }
+  if (n > 16) {
+    throw new Error('Number of public keys must be less than 17')
```

Comment:
Checking intepreter.cpp, it seems 20 is the consensus enforced limit:
https://github.com/bitcoin/bitcoin/blob/fc4f4547b7f6ea13d273ad926738fcdffee36295/src/script/interpreter.cpp#L914
https://github.com/bitcoin/bitcoin/blob/57b34599b2deb179ff1bd97ffeab91ec9f904d85/src/script/script.h#L28
Limited due to other policy reasons I guess, but for now I guess 20 since it's used by OP_CHECKMULTISIG?